data-processor/src/main/java/io/micronaut/data/processor/model/criteria/impl/SourcePersistentEntityCriteriaQueryImpl.java

```diff
@@ -88,7 +88,11 @@ public void visit(Predicate predicate) {
 
                 @Override
                 public void visit(PersistentPropertyPath<?> persistentPropertyPath) {
-                    result[0] = ((SourcePersistentPropertyPath) persistentPropertyPath).getProperty().getType().getName();
+                    ClassElement type = ((SourcePersistentPropertyPath) persistentPropertyPath).getProperty().getType();
```

Comment:
This is another change, in the example above
```
public abstract List<Book> findAuthorBooksById(@Id Long id);
```
result type comes out as java.util.List and we need actual type of the class in the list.

Reply:
The correct way is to check if it is SourceAssociation and extract the associated entity and get the type of it

Reply:
Updated, thanks!